### PR TITLE
engine: Fix version fetcher

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: March 14

- Fixed a bug in the version fetcher where we weren't awaiting the results from GitHub, so the fallback version was always used.
- Added a new variable for the 23.1 docs.

### CODE REVIEW

File: docs/get-started/quick-start.mdx [hunk 1]
```diff
@@ -154,7 +154,7 @@ so that the cluster can recover from a single-broker failure.
         - --mode dev-container
         # enable logs for debugging.
         - --default-log-level=debug
-      image: docker.redpanda.com/vectorized/redpanda:vVAR::REDPANDA_VERSION_22_3
+      image: docker.redpanda.com/vectorized/redpanda:vVAR::REDPANDA_VERSION_23_1
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:157]:
```suggestion
      image: docker.redpanda.com/redpandadata/redpanda:vVAR::REDPANDA_VERSION_23_1
```

**Review by Deflaimun — APPROVED** [review]
Suggestion (you don't have to implement if you don't want) 
In the code, we refer to the version variable as REDPANDA_VERSION (instead of using the number)

and at docusaurus.config.js L116 you define that if you don't fetch the version, you use the value from REDPANDA_VERSION_23_1

Other than that, the commands appear to be running fine

File: docusaurus.config.js [hunk 2]
```diff
@@ -113,8 +113,10 @@ module.exports = async () => {
                     // The fallback version to the right of the pipes (||)
                     // is used in cases where the version cannot be fetched from GitHub.
                     // You can assign the latest release data to any variable.
-                    REDPANDA_VERSION_22_3: latestRedpandaReleaseVersion || '22.3.11',
-                    REDPANDA_SHA_22_3: latestRedpandaReleaseCommitHash || '9eefb907c',
+                    REDPANDA_VERSION_23_1: latestRedpandaReleaseVersion || '23.1.1',
+                    REDPANDA_SHA_23_1: latestRedpandaReleaseCommitHash || '92761c',
```
Comment (shorgi) [inline docusaurus.config.js:117]: @JakeSCahill where is this commit SHA from? I can't find it when searching 
Github: https://github.com/search?q=org%3Aredpanda-data+92761c&type=commits

Comment (JakeSCahill) [reply docusaurus.config.js:117]: it's from the release tag: https://github.com/redpanda-data/redpanda/releases/tag/v23.1.1

**Comment by JakeSCahill** [comment]
> Suggestion (you don't have to implement if you don't want) In the code, we refer to the version variable as REDPANDA_VERSION (instead of using the number)
> 
> and at docusaurus.config.js L116 you define that if you don't fetch the version, you use the value from REDPANDA_VERSION_23_1
> 
> Other than that, the commands appear to be running fine

I'm not sure what you mean here @Deflaimun

File: docusaurus.config.js (hunk 2 above)
Comment (shorgi) [reply docusaurus.config.js:117]: Ah, thanks, it needs the last, seventh char.
```suggestion
                    REDPANDA_VERSION_23_1: latestRedpandaReleaseVersion || '23.1.1',
                    REDPANDA_SHA_23_1: latestRedpandaReleaseCommitHash || '92761c3',
```

**Review by shorgi — APPROVED** [review]

**Pull request merged**